docs/application/web/guides/machine-learning/single.md

```diff
@@ -0,0 +1,153 @@
+
+## Open the model
+
+1. To load the model from storage, use `openModel`:
+```javascript
+  var model = tizen.ml.single.openModel("documents/mobilenet_v1_1.0_224_quant.tflite");
+```
+
+2. If the model supports dynamic tensors, you can:
+
+- provide custom input and output tensors:
+```javascript
+  var input = new tizen.ml.TensorsInfo()
+  input.addTensorInfo("input", "FLOAT32", [3])
+  var output = new tizen.ml.TensorsInfo()
+  output.addTensorInfo("output", "FLOAT32", [3])
+  var model = tizen.ml.single.openModel("documents/dynamic.tflite", input, output);
+  input.dispose();
+  output.dispose();
+```
+
+- set the model's `input` tensor:
+```javascript
+  var input = new tizen.ml.TensorsInfo()
+  input.addTensorInfo("input", "FLOAT32", [3])
+  model.input = input;
+  input.dispose();
+```
+
+- open model with dynamic mode, which allows you to [invoke mode](#invoke) with variable tensors.
+
+```javascript
+  var model = tizen.ml.single.openModel("documents/mobilenet_v1_1.0_224_quant.tflite", null, null, "ANY", "ANY", true);
+```
+
+3. You can also provide parameters that will define framework and hardware for your model:
+
+```javascript
+  var model = tizen.ml.single.openModel("documents/mobilenet_v1_1.0_224_quant.tflite", null, null, "TENSORFLOW_LITE", "ANY");
+```
+
+4. Always close SingleShot objects when you don't need them anymore:
+
+```javascript
+  model.close()
+```
+
+5. You can also open model asyncronously with `openModelAsync`.
+
+This approach avoids blocking UI when opening big model file.
+
+```javascript
+  function errorCallback(error)
+  {
+    console.log("Error while opening model: " + error.message);
+  }
+
+  function successCallback(model)
+  {
+    console.log("Model opened successfully");
+
+    // Do inference here
+
+    model.close();
+  }
+  tizen.ml.single.openModelAsync("documents/mobilenet_v1_1.0_224_quant.tflite", successCallback, errorCallback);
+```
+
+## Invoke
+To invoke the neural network model, you need to create the `TensorsData` instance to pass the input data to the model.
+You can find the details about using `TensorsData` interface in [Tensors Management](./tensors.md) guide.
+
+1. Create TensorsData object based on information from model:
+
+```javascript
+  var inputTensorsInfo = model.input;
+  var inputTensorsData = inputTensorsInfo.getTensorsData();
+```
+
+2. Set the raw data to `TensorsData` object:
+
+```javascript
+  // create or fetch data
+  var inputData = new Uint8Array(224 * 224 * 3);
+  inputTensorsData.setTensorRawData(0, inputData);
+```
+
+3. Invoke the model:
+
+```javascript
+  var result = model.invoke(inputTensorsData)
+```
+
+Computations in more complex models may take long time, to limit this time, you can use `setTimeout`:
+
+```javascript
+  model.setTimeout(100);
+```
+
+This will cause `TimeoutError` in case of invoking for longer that 100 ms.
```

Comment:
Consider adding the information about time units (milliseconds) in the sentence above the code snippet.

Reply:
Done.